modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/ResolverSetResult.java

```diff
@@ -0,0 +1,189 @@
+         * @throws IllegalArgumentException is {@code parameter} is {@code null}
+         */
+        public Builder add(Parameter parameter, Object value)
+        {
+            checkArgument(parameter != null, "parameter cannot be null");
+            values.put(parameter, value);
+            hashCode = 31 * hashCode + (value == null ? 0 : value.hashCode());
+            return this;
+        }
+
+        /**
+         * Creates a new {@link ResolverSetResult}
+         *
+         * @return the build instance
+         */
+        public ResolverSetResult build()
+        {
+            return new ResolverSetResult(Collections.unmodifiableMap(values), hashCode);
+        }
+    }
+
+    /**
+     * Creates a new {@link Builder} instance. You should use a new builder
+     * per each instance you want to create
+     *
+     * @return a {@link Builder}
+     */
+    public static Builder newBuilder()
+    {
+        return new Builder();
+    }
+
+    private final Map<Parameter, Object> evaluationResult;
+    private final Map<String, Object> parameterName2Result;
+    private final int hashCode;
+
+    private ResolverSetResult(Map<Parameter, Object> evaluationResult, int hashCode)
+    {
+        this.evaluationResult = evaluationResult;
+        parameterName2Result = new HashMap<>(evaluationResult.size());
+        for (Map.Entry<Parameter, Object> entry : evaluationResult.entrySet())
+        {
+            parameterName2Result.put(entry.getKey().getName(), entry.getValue());
+        }
+
+        this.hashCode = hashCode;
+    }
+
+    /**
+     * Returns the value associated with the given {@code parameter}
+     *
+     * @param parameter a {@link Parameter} which was registered with the builder
+     * @return the value associated to that {@code parameter}
+     * @throws NoSuchElementException if the {@code parameter} has not been registered through the builder
+     */
+    public Object get(Parameter parameter)
+    {
+        if (!evaluationResult.containsKey(parameter))
+        {
+            throw new NoSuchElementException("This result contains no information for the parameter: " + parameter.getName());
+        }
+
+        return evaluationResult.get(parameter);
+    }
+
+    /**
+     * Returns the value associated with the {@link Parameter} of the given {@code parameterName}
+     *
+     * @param parameterName the name of the {@link Parameter} which value you seek
+     * @return the value associated to that {@code parameterName}
+     * @throws NoSuchElementException if the {@code parameterName} has not been registered through the builder
+     */
+    public Object get(String parameterName)
+    {
+        if (!parameterName2Result.containsKey(parameterName))
+        {
+            throw new NoSuchElementException("This result contains no information for the parameter: " + parameterName);
+        }
+
+        return parameterName2Result.get(parameterName);
+    }
+
+    public Map<Parameter, Object> asMap()
```

Comment:
Because there're cases in which it's necessary to navigate the results in a key/value fashion, and exposing it as map is more useful than exposing a list of keys and then using the get(String) method